dwitter/serializers_v2.py

```diff
@@ -8,11 +8,53 @@
 
 class UserSerializer(serializers.ModelSerializer):
     avatar = serializers.SerializerMethodField()
+    # Confirmation password, only for validation
+    password2 = serializers.CharField(allow_blank=False, write_only=True)
 
     class Meta:
         model = User
-        read_only_fields = ('id', 'username', 'avatar')
-        fields = read_only_fields
+        fields = ('id', 'avatar', 'username', 'password', 'password2', 'email')
+        extra_kwargs = {
+            'id': {'read_only': True},
+            'avatar': {'read_only': True},
+            'password': {'write_only': True},
+            'password2': {'write_only': True},
+            # email is not required in the database, so need to be set explicitly here
+            'email': {'write_only': True, 'required': True,
+                      'allow_null': False, 'allow_blank': False}
+        }
+
+    def validate(self, data):
+        """
+        Checks to be sure that the received password and password2 are the same
+        And that the password meets auth.password_validation
+        """
+        errors = dict()
+        try:
+            # Note: 'password2' gets removed from the data here with pop
+            if data['password'] != data.pop('password2'):
+                errors['password'] = ('Passwords do not match')
```

Comment:
What's up with the parens here? If `errors['password']` should be a tuple of strings, then you need an extra comma: `('Passwords do not match',)`. If it should just be a string, we can remove the parens so that it doesn't look like an attempt at making up a tuple.